apps/atomy-q/API/app/Adapters/InsightOperations/ProviderInsightNarrativeAdapter.php

```
@@ -0,0 +1,111 @@
+            provenance: $this->provenance($response, $payload, $actorId),
+        );
+    }
+
+    /**
+     * @param array<string, mixed> $response
+     * @return array<string, mixed>
+     */
+    private function unwrapPayload(array $response): array
+    {
+        if (isset($response['payload']) && ! is_array($response['payload'])) {
+            throw new \DomainException('AI provider returned a non-array payload.');
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '49', column '23'). (undefined)

(MissingImport)